src/connectors/onchain_events/mod.rs

```diff
@@ -167,6 +169,7 @@ impl Subscriber {
             start_block_number: config.start_block_number,
             stop_block_number: config.stop_block_number,
             statsd_client,
+            timestamp_cache: Mutex::new(HashMap::new()),
```

Comment:
The cache should have some max capacity so we don't leak memory. You can use the moka library for an lru cache.